libvast/src/system/configuration.cpp

```diff
@@ -229,6 +246,11 @@ caf::error configuration::parse(int argc, char** argv) {
   }
   // Flatten everything for simplicity.
   merged_config = flatten(merged_config);
+  // Overwrite current config with VAST_ environment variables, potentially
+  // appending new config keys at the end.
+  for (const auto& [key, value] : detail::environment())
+    if (key.starts_with("VAST_") && !value.empty())
+      merged_config[env_to_config_key(key)] = std::string{value};
```

Comment:
Should the value be put through the `data` parser as an approximation here?

But even if you do this, I don't think the following will work:

```bash
# Works as expected, parses a list of strings:
vast start --commands='spawn source pcap'

# Does not work, parses as a single string
VAST_START__COMMANDS='spawn source pcap' vast start
```

Reply:
Not sure what you mean. Something like `VAST_EXPORT__TIMEOUT=1ns ./bin/vast -N export json 1.2.3.4` correctly parses the env variable as `duration`.

Reply:
But the env variable should be a string, not a duration, because we expect a string for that option.

Reply:
Why does it work then?

Reply:
Do we maybe parse as a data duration, and then when converting into a caf::settings we convert it back to a string, which we then convert back to a data string and then parse as a duration?

Reply:
You're right, "transparent lists" don't work.